### ISSUE TITLE
[PostgreSQL] Cannot create or update row with large Decimal value within `numeric` range

### ISSUE THREAD

**Issue by codazzo** [prisma/prisma#11464]
### Bug description

I have a schema with a model which has a `Decimal` value. When I try to create or update an entry passing as value a large number such as `Prisma.Decimal('93431006234343600001')`, the operation fails with the following error

```
  Error parsing GraphQL query: query parse error: Parse error at 3:12
number too large to fit in target type
```

### How to reproduce

I have created a reproduction repo which can be found at https://github.com/codazzo/prisma-decimal-precision-repro

### Expected behavior

My understanding is that with a `Decimal` type I should be able to run

```ts
  await prisma.entry.create({
    data: {
      value: new Prisma.Decimal('93431006234343600001')
    }
  })
```

Note that I _am_ able to `find` any entities that hold large values from the DB.
If the value is `93431006234343600001` in the DB, I can successfully call `prisma.entry.findUnique()` and get the value back.

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model Entry {
  id    Int     @id @default(autoincrement())
  value Decimal
}
```

### Environment & setup

OS: macOS Big Sur 11.6.2 Kernel Version 20.6.0
Database: PostgreSQL 14.1
Node version: v16.9.1

### Prisma Version

```
prisma                  : 3.8.1
@prisma/client          : 3.8.1
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine 34df67547cf5598f5a6cd3eb45f14ee70c3fb86f (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Migration Engine        : migration-engine-cli 34df67547cf5598f5a6cd3eb45f14ee70c3fb86f (at node_modules/@prisma/engines/migration-engine-darwin-arm64)
Introspection Engine    : introspection-core 34df67547cf5598f5a6cd3eb45f14ee70c3fb86f (at node_modules/@prisma/engines/introspection-engine-darwin-arm64)
Format Binary           : prisma-fmt 34df67547cf5598f5a6cd3eb45f14ee70c3fb86f (at node_modules/@prisma/engines/prisma-fmt-darwin-arm64)
Default Engines Hash    : 34df67547cf5598f5a6cd3eb45f14ee70c3fb86f
Studio                  : 0.452.0
```

**Comment by diwu1989** [comment]
I'm hitting this problem too, and with BigInt column type.

Please add some basic test cases for this and fix it asap, basically all big number support is broken.

**Comment by shortcircuit3** [comment]
Any update here? I am completely blocked from using large numbers with prisma.

**Comment by sameghosts** [comment]
Do we have any update on this or a progress update here? A project I am working on is blocked by this.

**Comment by shortcircuit3** [comment]
It looks like this works: https://github.com/codazzo/prisma-decimal-precision-repro/pull/1/files

Will test more and follow up...

**Comment by shortcircuit3** [comment]
I was suggested to save the big number on create as a string and that seems to work, but updating values still doesnt work

this works for creation
```js
  await prisma.entry.create({
    data: {
      value: '93431006234343600001111111',
    },
  });
```

when updating it doesnt work
```js
await prisma.entry.update({
    where: {
      id: 1,
    },
    data: {
      value: 123861532518063870000,
    },
  });
```

I'm not sure how to get updating to work because I'm getting type errors when i try to force it to be a string

My reproduction: https://github.com/shortcircuit3/prisma-decimal-precision-repro/blob/master/index.ts

cc @ruheni

**Comment by ruheni** [comment]
You can use the `String()` method to cast the value to a string. It works for both `create` and `update` operations. The update operation in the seems to be number. 🤔

```ts
await prisma.entry.update({
    where: {
      id: 1,
    },
    data: {
      value: String(123861532518063870000),
    },
  });
```

**Comment by shortcircuit3** [comment]
@ruheni this doesn't work because of types. Any idea?

```
Type 'string' is not assignable to type 'number | FloatFieldUpdateOperationsInput | undefined'.ts(2322)

index.d.ts(1637, 5): The expected type comes from property 'value' which is declared here on type '(Without<EntryUpdateInput, EntryUncheckedUpdateInput> & EntryUncheckedUpdateInput) | (Without<...> & EntryUpdateInput)'
```

<img width="574" alt="image" src="https://user-images.githubusercontent.com/421362/167658810-1a309ac0-19e3-47c8-97ab-f8720fa9defe.png">

**Comment by shortcircuit3** [comment]
Yes, the problem exists for float too...

**Comment by shortcircuit3** [comment]
Do you think you'll ever support `float8` or is moving to `numeric` the only thing we can do?

**Comment by shortcircuit3** [comment]
@ruheni Any update?

**Comment by Jolg42** [comment]
Trying in our internal dev version `4.4.0-dev.25`, it seems to be fixed.

**Comment by SevInf** [comment]
Yes, this case is fixed by #15157 and should go out with 4.4 release. Other decimal precision issues are tracked in #8160

**Comment by janpio** [comment]
For internal bookkeeping: This was considered a duplicate of https://github.com/prisma/prisma/issues/5925
